### PR TITLE
add e2e test for search settings

### LINKED ISSUE #12868: Create e2e test for new search feature in Settings component
Acceptance Criteria:
- Add new e2e test that covers simple use case of the new search feature implemented in the search component

Related task: [https://github.com/MetaMask/metamask-extension/issues/12760](https://github.com/MetaMask/metamask-extension/issues/12760)

### PR DESCRIPTION
**Pull request by mirjanaKukic**
Resolves [#13396](https://github.com/MetaMask/metamask-extension/issues/12868)
This covers features merged in https://github.com/MetaMask/metamask-extension/pull/13730

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
hey @danjm , I noticed that there is the same ticket created for search feature : https://github.com/MetaMask/metamask-extension/issues/14190 
Could you please take a look? Thanks.

**Comment by seaona** [comment]
hi @mirjanaKukic, @PeterYinusa also noticed that this PR is tackling the same functionality as this one https://github.com/MetaMask/metamask-extension/pull/14259 I wrote.
Unfortunately, the same thing happened with Token Details testcase: https://github.com/MetaMask/metamask-extension/pull/13666 and https://github.com/MetaMask/metamask-extension/pull/13818/.

Any ideas on how could we better coordinate so we do not duplicate work? I guess it's a good indicator that we are missing some awareness on which testing activities are happening and by whom. Could we maybe use the QA dashboard for improving that process? @hilvmason @danjm

File: test/e2e/tests/settings.spec.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        });
+        assert.equal(await settingsButton.getText(), 'Settings');
+        await settingsButton.click();
+
+        await driver.findElement('#search-settings');
+
+        // enters expected search
+        await driver.fill('#search-settings', 'advanced');
+
+        const searchResult = await driver.findElements(
```
Comment (PeterYinusa) [inline test/e2e/tests/settings.spec.js:40]: Should this be searchResults (plural)

File: test/e2e/tests/settings.spec.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // goes to settings page
+        await driver.clickElement('.account-menu__icon');
+        const settingsButton = await driver.findClickableElement({
+          tag: 'div',
+          text: 'Settings',
+        });
+        assert.equal(await settingsButton.getText(), 'Settings');
```
Comment (PeterYinusa) [inline test/e2e/tests/settings.spec.js:32]: I don't think this assertion is necessary, since we are finding the element using the text `Settings`

**Comment by PeterYinusa** [comment]
> hi @mirjanaKukic, @PeterYinusa also noticed that this PR is tackling the same functionality as this one #14259 I wrote. Unfortunately, the same thing happened with Token Details testcase: #13666 and #13818.
> 
> Any ideas on how could we better coordinate so we do not duplicate work? I guess it's a good indicator that we are missing some awareness on which testing activities are happening and by whom. Could we maybe use the QA dashboard for improving that process? @hilvmason @danjm

This PR looks good, just a few small comments on improvements that can be made. I think this PR should be closed in favour of https://github.com/MetaMask/metamask-extension/pull/14259, as #14259 has more coverage

**Comment by mirjanaKukic** [comment]
@darkwing This one should be closed since it is duplicate of https://github.com/MetaMask/metamask-extension/pull/14259

**Pull request closed without merging**